### PR TITLE
installer: use steps for e2e-{aws,azure,gcp}-shared-vpc

### PR DESCRIPTION
**Pull request by abhinavdahiya**

### CODE REVIEW

File: ci-operator/step-registry/ipi/aws/pre/sharednetwork/OWNERS [hunk 1]
```diff
@@ -0,0 +1,11 @@
+approvers:
```
Comment (wking) [inline ci-operator/step-registry/ipi/aws/pre/sharednetwork/OWNERS:1]: should probably include @jstuever and @patrickdillon based on [installer AWS approvers][1].  And possibly some folks in your current list aren't interested in being approvers for this specific directory (folks with a general interest in multi-step are probably already approvers from root-wards directories and don't need to be repeated down in this one).

[1]: https://github.com/openshift/installer/blob/145e5ccdf32513ee96c777853e1a7a7620781960/OWNERS_ALIASES#L52-L53

File: ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-commands.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
```
Comment (wking) [inline ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-commands.sh:8]: Don't we have an image that contains `yq` now?  Alternative would be to have an image that contains `jq` and write the install-config in JSON.

File: ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-ref.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+ref:
+  as: ipi-conf-aws-sharednetwork
```
Comment (wking) [inline ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-ref.yaml:2]: We do this in a few places already, but I [hear][1] there are some ci-tools issues with using the same name for both a `ref` and a `chain`.  Might be worth using different names for them here; or maybe that's no longer an issue.

[1]: https://github.com/openshift/release/pull/7931#issuecomment-604144637

File: ci-operator/step-registry/ipi/conf/azure/sharednetwork/OWNERS [hunk 4]
```diff
@@ -0,0 +1,2 @@
+approvers:
+- abhinavdahiya
```
Comment (wking) [inline ci-operator/step-registry/ipi/conf/azure/sharednetwork/OWNERS:2]: Can we extend this to [include][1] @fabianofranz and @jhixson74 ?  Having a single approver seems like a risk for over-silo'ed understanding (although root-ward approvers are also theoretically on the hook to provide maintenance in this subdir as well).

[1]: https://github.com/openshift/installer/blob/145e5ccdf32513ee96c777853e1a7a7620781960/OWNERS_ALIASES#L63-L65

File: ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-ref.yaml (hunk 3 above)
Comment (bbguimaraes) [reply ci-operator/step-registry/ipi/conf/aws/sharednetwork/ipi-conf-aws-sharednetwork-ref.yaml:2]: https://github.com/openshift/ci-tools/pull/818

**Comment by wking** [comment]
[GCP][1]:

```
 + go build -mod=vendor -ldflags ' -X github.com/openshift/installer/pkg/version.Raw=unreleased-master-3184-gda3ca1e9b14214fc7a2fb71d1c6edbf45a764f3b-dirty -X github.com/openshift/installer/pkg/version.Commit=da3ca1e9b14214fc7a2fb71d1c6edbf45a764f3b -s -w' -tags ' release' -o bin/openshift-install ./cmd/openshift-install
error: build error: no such image 
```

Dunno what that's about.

[1]: https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/9517/rehearse-9517-pull-ci-openshift-installer-master-e2e-gcp-shared-vpc/5

**Comment by wking** [comment]
[AWS][1] lease exhaustion (`failed to acquire lease: resources not found`).  [Azure][2] died on install (`OSProvisioningTimedOut`).  Maybe some cool-down time will help, and things will green up when the retest bot shows up later tonight.

[1]: https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/9517/rehearse-9517-pull-ci-openshift-installer-master-e2e-aws-shared-vpc/7
[2]: https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_release/9517/rehearse-9517-pull-ci-openshift-installer-master-e2e-azure-shared-vpc/6

File: ci-operator/step-registry/ipi/conf/azure/sharednetwork/OWNERS (hunk 4 above)
Comment (wking) [reply ci-operator/step-registry/ipi/conf/azure/sharednetwork/OWNERS:2]: Out of band, @abhinavdahiya wants to punt on this too, like [here][1].

[1]: https://github.com/openshift/release/pull/9517#discussion_r438269592

**Pull request merged**